### PR TITLE
Write docs to gh-pages branch.

### PR DESCRIPTION
**Pull request by adarob**

### CODE REVIEW

File: music/scripts/generate-docs.sh [hunk 1]
```diff
@@ -16,14 +16,24 @@
 
 # Regenerates the docs
 #
-# To run, execute 'yarn doc'.
+# To run, execute 'yarn docs'.
+
+# Exit on error.
+set -e
 
 # Generate the docs.
```
Comment (notwaldorf) [inline music/scripts/generate-docs.sh]: technically this comment should go above the npx line, these first lines are just setting up variables

File: music/scripts/generate-docs.sh [hunk 2]
```diff
@@ -49,3 +59,16 @@ for path in ./*.html; do
     sed -i "" "s%${search}%${replace}%g" $path
   fi
 done
+
+cd $currDir
+yarn build-demos
+mkdir -p $TMP_DIR/demos && cp demos/*.{js,html,mid,css} $TMP_DIR/demos
+git checkout gh-pages
```
Comment (notwaldorf) [inline music/scripts/generate-docs.sh]: Maybe a quick summary of what this is doing?

File: music/scripts/generate-docs.sh [hunk 3]
```diff
@@ -49,3 +59,16 @@ for path in ./*.html; do
...
+docsDir=$(git rev-parse --show-toplevel)
+cd $baseDir
+git rm -fr music
+rsync -a $TMP_DIR/ music/
```
Comment (notwaldorf) [inline music/scripts/generate-docs.sh]: why rsync and not something like cp?

File: music/scripts/generate-docs.sh [hunk 4]
```diff
@@ -49,3 +59,16 @@ for path in ./*.html; do
...
+git add music
+git commit -m "Updating music docs: $currDate"
```
Comment (notwaldorf) [inline music/scripts/generate-docs.sh]: i would create that currDate variable here, not up top (i forgot what it did already)

File: music/scripts/generate-docs.sh [hunk 5]
```diff
@@ -16,14 +16,24 @@
...
-# To run, execute 'yarn doc'.
+# To run, execute 'yarn docs'.
+
+# Exit on error.
+set -e
 
 # Generate the docs.
-npx typedoc --tsconfig "tsconfig.es5.json" --sourcefile-url-prefix 'https://github.com/tensorflow/magenta-js/tree/master/music/src/' --out ../docs/music src --mode modules --excludePrivate --exclude '**/*+(index|test|lib).ts' --excludeExternals
+TMP_DIR=/tmp/mm_docs
+currRepo=$(git rev-parse --abbrev-ref HEAD)
```
Comment (notwaldorf) [inline music/scripts/generate-docs.sh]: i think you mean currBranch not repo, right?

File: music/scripts/generate-docs.sh (hunk 3 above)
Comment (adarob) [reply music/scripts/generate-docs.sh]: So that we don't clobber untracked files from the other branch.

**Pull request merged**